src/resource.py

```diff
@@ -288,6 +288,17 @@ def update_cluster_id(self, id_: str) -> None:
         """Update clusterID."""
         self.update({"parameters": {"clusterID": id_}})
 
+    def set_default(self, is_default: bool = True) -> None:
+        """Set default status of the StorageClass."""
+        patch = {
+            "metadata": {
+                "annotations": {
+                    "storageclass.kubernetes.io/is-default-class": str(is_default).lower()
+                }
+            }
+        }
+        self.update(patch)
```

Comment:
Would there be any value in checking the current value to not apply a value that is already set (i.e. disable `is-default-class` when it is already disabled)?

Reply:
That would require making calls to k8s api to determine if the current SC is default or not and that's slower than current implementation.

Alternatively we can track the "default" status of each SC internally in `StoredState` but that might not be always reliable.